source/postprocess/geoid.cc

```diff
@@ -259,20 +246,99 @@ namespace aspect
             // spherical harmonic expansion compared to using just
             // one average value per face, especially in the coarse
             // meshes.
-            fe_face_values[this->introspection().extractors.temperature].get_function_values(topo_vector, topo_values);
 
-            // if the cell at top boundary, add its contributions dynamic topography storage vector
-            if (face_idx != numbers::invalid_unsigned_int && at_upper_surface)
+            if (include_surface_topo_contribution == true)
               {
-                for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
-                  surface_stored_values.emplace_back (std::make_pair(fe_face_values.quadrature_point(q), std::make_pair(fe_face_values.JxW(q), topo_values[q])));
+                // If the cell is at the top boundary, add its contributions to the topography surface storage vector.
+                if (at_upper_surface)
```

Comment:
I think this function would be easier if you reverted the order of these if-conditions. Currently you have the following:
```
if (include_contribution){
  if (at_upper_surface){
    if (use_real_surface)
    else
  }
}
else
  if (at_upper_surface){
}
```
This would be simpler if you would do:
```
if (at_upper_surface){
  if (include_contribution)
    if (use_real_surface)
    else
  else
}
```

And the same for the CMB block below.